pkg/apis/core/validation/validation.go

```diff
@@ -3774,7 +3774,7 @@ func podSpecHasContainer(spec *core.PodSpec, containerName string) bool {
 
 const (
 	// a sysctl segment regex, concatenated with dots to form a sysctl name
-	SysctlSegmentFmt string = "[a-z0-9]([-_a-z0-9]*[a-z0-9])?"
+	SysctlSegmentFmt string = "[a-z0-9]([-_/a-z0-9]*[a-z0-9])?"
```

Comment:
Relaxing validation of API fields has to be done over two releases so that we don't allow data to be persisted that the prior version of the API server considered invalid. That breaks multi-server clusters that are in the middle of an upgrade and have an API server at version N that allows data to be persisted with the relaxed validation, but still has an API server at version N-1 that rejects an update to that object because it is still enforcing the stricter validation.

Reply:
In release 1:
* use strict validation for Create requests, and for Update requests where the existing object's sysctl does not contain a slash
* use relaxed validation for Update requests where the existing object's sysctl contains a slash

In release 2:
* use relaxed validation everywhere

This implies all the places we do validation know whether to apply strict or relaxed validation, and that we determine that based on existing data in Update requests. See construction of PodValidationOptions in pkg/api/pod/util.go, and propagation to all pod/podspec validation methods for a pattern for handling this

Reply:
oh I see now and it's better to deal with it this way. I'll update that, thanks!